src/bootstrap/src/lib.rs

```diff
@@ -488,7 +488,7 @@ impl Build {
         }
         self.config.update_submodule(submodule);
         let absolute_path = self.config.src.join(submodule);
-        if dir_is_empty(&absolute_path) {
+        if !absolute_path.exists() || dir_is_empty(&absolute_path) {
```

Comment:
I am unsure if this is needed since we expect submodules to be present?

Reply:
It would just produce a nicer error message in the original issue (since the directory was missing). If a submodule directory is ever removed again in the future, it would make it a bit simpler to debug (I didn't learn much from the original error). But if you think it's not needed, I'll drop the first commit.

Reply:
Sure it makes sense